lib/linter/linter.js

```diff
@@ -1119,6 +1119,10 @@ function runRules(sourceCode, configuredRules, ruleMapper, parserName, languageO
             };
         }
 
+        if (typeof ruleListeners !== "object" || ruleListeners === null) {
```

Comment:
```suggestion
        if (typeof ruleListeners === "undefined" || ruleListeners === null) {
```

This would still fail in some cases where it didn't before, like `return 0`.